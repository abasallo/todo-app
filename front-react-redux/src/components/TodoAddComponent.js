import React from 'react'
import { Component, PropTypes } from 'react'
import { FloatingActionButton, FlatButton, Dialog, TextField } from 'material-ui'
import Add from 'react-material-icons/icons/content/add'

import config from '../config'

class TodoAddComponent extends Component {

    state = { open: false, text: '' }

    actions = [
        <FlatButton label='Cancel' primary={ true } onClick={ () => this.setState({open: false}) }/>,
        <FlatButton label='Submit' primary={ true } onClick={ () => {
            this.props.onAddTodo(this.state.text)
            this.setState({ open: false })}}/>]

    render() { return(
        <div>
            <FloatingActionButton style={ config.components.todoAddButtonStyle } zDepth={ 1 } mini={ true } onClick={ () => this.setState({ open: true }) }>
                <Add/>
            </FloatingActionButton>

            <Dialog title={ config.components.todoAddCaption } actions={ this.actions } modal={ true } open={ this.state.open }>
                <TextField hintText={ config.todo.defaults.text } onChange={ (e) => this.setState({ text: e.target.value }) }/>
            </Dialog>
        </div>
    )}
}

TodoAddComponent.propTypes = { onAddTodo: PropTypes.func.isRequired }

export default TodoAddComponent
