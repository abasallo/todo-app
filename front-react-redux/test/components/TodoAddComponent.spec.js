import React from 'react'
import { shallow } from 'enzyme'

import TodoAddComponent from '../../src/components/TodoAddComponent'

it('renders properly', () => { expect(shallow(<TodoAddComponent onAddTodo={ () => {} }/>)).toMatchSnapshot() })
