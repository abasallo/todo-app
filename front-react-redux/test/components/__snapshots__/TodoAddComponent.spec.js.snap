exports[`test renders properly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <FloatingActionButton
        disabled={false}
        mini={true}
        onClick={[Function]}
        secondary={false}
        style={
            Object {
                "float": "right",
                "marginBottom": "1.5%",
                "marginRight": "1%",
              }
        }
        zDepth={1}>
        <ContentAdd />
    </FloatingActionButton>
    <Dialog
        actions={
            Array [
                <FlatButton
                  disabled={false}
                  fullWidth={false}
                  label="Cancel"
                  labelPosition="after"
                  labelStyle={Object {}}
                  onClick={[Function]}
                  onKeyboardFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onTouchStart={[Function]}
                  primary={true}
                  secondary={false} />,
                <FlatButton
                  disabled={false}
                  fullWidth={false}
                  label="Submit"
                  labelPosition="after"
                  labelStyle={Object {}}
                  onClick={[Function]}
                  onKeyboardFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onTouchStart={[Function]}
                  primary={true}
                  secondary={false} />,
              ]
        }
        autoDetectWindowHeight={true}
        autoScrollBodyContent={false}
        modal={true}
        open={false}
        repositionOnUpdate={true}
        title="Add TODO">
        <TextField
            disabled={false}
            floatingLabelFixed={false}
            fullWidth={false}
            hintText="New Task"
            multiLine={false}
            onChange={[Function]}
            rows={1}
            type="text"
            underlineShow={true} />
    </Dialog>
</div>,
  "nodes": Array [
    <div>
      <FloatingActionButton
            disabled={false}
            mini={true}
            onClick={[Function]}
            secondary={false}
            style={
                  Object {
                        "float": "right",
                        "marginBottom": "1.5%",
                        "marginRight": "1%",
                      }
            }
            zDepth={1}>
            <ContentAdd />
      </FloatingActionButton>
      <Dialog
            actions={
                  Array [
                        <FlatButton
                          disabled={false}
                          fullWidth={false}
                          label="Cancel"
                          labelPosition="after"
                          labelStyle={Object {}}
                          onClick={[Function]}
                          onKeyboardFocus={[Function]}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          onTouchStart={[Function]}
                          primary={true}
                          secondary={false} />,
                        <FlatButton
                          disabled={false}
                          fullWidth={false}
                          label="Submit"
                          labelPosition="after"
                          labelStyle={Object {}}
                          onClick={[Function]}
                          onKeyboardFocus={[Function]}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          onTouchStart={[Function]}
                          primary={true}
                          secondary={false} />,
                      ]
            }
            autoDetectWindowHeight={true}
            autoScrollBodyContent={false}
            modal={true}
            open={false}
            repositionOnUpdate={true}
            title="Add TODO">
            <TextField
                  disabled={false}
                  floatingLabelFixed={false}
                  fullWidth={false}
                  hintText="New Task"
                  multiLine={false}
                  onChange={[Function]}
                  rows={1}
                  type="text"
                  underlineShow={true} />
      </Dialog>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <TodoAddComponent
        onAddTodo={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": TodoAddComponent {
        "_reactInternalInstance": [Circular],
        "actions": Array [
          <FlatButton
            disabled={false}
            fullWidth={false}
            label="Cancel"
            labelPosition="after"
            labelStyle={Object {}}
            onClick={[Function]}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            primary={true}
            secondary={false} />,
          <FlatButton
            disabled={false}
            fullWidth={false}
            label="Submit"
            labelPosition="after"
            labelStyle={Object {}}
            onClick={[Function]}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            primary={true}
            secondary={false} />,
        ],
        "context": Object {},
        "props": Object {
          "onAddTodo": [Function],
        },
        "refs": Object {},
        "state": Object {
          "open": false,
          "text": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <FloatingActionButton
                    disabled={false}
                    mini={true}
                    onClick={[Function]}
                    secondary={false}
                    style={
                              Object {
                                        "float": "right",
                                        "marginBottom": "1.5%",
                                        "marginRight": "1%",
                                      }
                    }
                    zDepth={1}>
                    <ContentAdd />
          </FloatingActionButton>
          <Dialog
                    actions={
                              Array [
                                        <FlatButton
                                          disabled={false}
                                          fullWidth={false}
                                          label="Cancel"
                                          labelPosition="after"
                                          labelStyle={Object {}}
                                          onClick={[Function]}
                                          onKeyboardFocus={[Function]}
                                          onMouseEnter={[Function]}
                                          onMouseLeave={[Function]}
                                          onTouchStart={[Function]}
                                          primary={true}
                                          secondary={false} />,
                                        <FlatButton
                                          disabled={false}
                                          fullWidth={false}
                                          label="Submit"
                                          labelPosition="after"
                                          labelStyle={Object {}}
                                          onClick={[Function]}
                                          onKeyboardFocus={[Function]}
                                          onMouseEnter={[Function]}
                                          onMouseLeave={[Function]}
                                          onTouchStart={[Function]}
                                          primary={true}
                                          secondary={false} />,
                                      ]
                    }
                    autoDetectWindowHeight={true}
                    autoScrollBodyContent={false}
                    modal={true}
                    open={false}
                    repositionOnUpdate={true}
                    title="Add TODO">
                    <TextField
                              disabled={false}
                              floatingLabelFixed={false}
                              fullWidth={false}
                              hintText="New Task"
                              multiLine={false}
                              onChange={[Function]}
                              rows={1}
                              type="text"
                              underlineShow={true} />
          </Dialog>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <FloatingActionButton
                    disabled={false}
                    mini={true}
                    onClick={[Function]}
                    secondary={false}
                    style={
                              Object {
                                        "float": "right",
                                        "marginBottom": "1.5%",
                                        "marginRight": "1%",
                                      }
                    }
                    zDepth={1}>
                    <ContentAdd />
          </FloatingActionButton>
          <Dialog
                    actions={
                              Array [
                                        <FlatButton
                                          disabled={false}
                                          fullWidth={false}
                                          label="Cancel"
                                          labelPosition="after"
                                          labelStyle={Object {}}
                                          onClick={[Function]}
                                          onKeyboardFocus={[Function]}
                                          onMouseEnter={[Function]}
                                          onMouseLeave={[Function]}
                                          onTouchStart={[Function]}
                                          primary={true}
                                          secondary={false} />,
                                        <FlatButton
                                          disabled={false}
                                          fullWidth={false}
                                          label="Submit"
                                          labelPosition="after"
                                          labelStyle={Object {}}
                                          onClick={[Function]}
                                          onKeyboardFocus={[Function]}
                                          onMouseEnter={[Function]}
                                          onMouseLeave={[Function]}
                                          onTouchStart={[Function]}
                                          primary={true}
                                          secondary={false} />,
                                      ]
                    }
                    autoDetectWindowHeight={true}
                    autoScrollBodyContent={false}
                    modal={true}
                    open={false}
                    repositionOnUpdate={true}
                    title="Add TODO">
                    <TextField
                              disabled={false}
                              floatingLabelFixed={false}
                              fullWidth={false}
                              hintText="New Task"
                              multiLine={false}
                              onChange={[Function]}
                              rows={1}
                              type="text"
                              underlineShow={true} />
          </Dialog>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TodoAddComponent
    onAddTodo={[Function]} />,
}
`;
